src/Core/Settings.h

```diff
@@ -325,6 +325,7 @@ class IColumn;
     \
     M(Bool, join_use_nulls, false, "Use NULLs for non-joined rows of outer JOINs for types that can be inside Nullable. If false, use default value of corresponding columns data type.", IMPORTANT) \
     \
+    M(Int32, join_output_by_rowlist_perkey_rows_threshold, 5, "The lower limit of per-key average rows in the right table to determine whether to output by row list in hash join.", 0) \
```

Comment:
The setting is used as unsigned int (size_t) but declared as Int32 here. It makes sense to make it unsigned (before 24.9)

Reply:
#69897